src/lib.rs

```diff
@@ -0,0 +1,886 @@
+		let (_channel_manager_blockhash, channel_manager) = {
+			if let Ok(mut f) = fs::File::open(format!("{}/manager", ldk_data_dir.clone())) {
+				let mut channel_monitor_mut_references = Vec::new();
+				for (_, channel_monitor) in channel_monitors.iter_mut() {
+					channel_monitor_mut_references.push(channel_monitor);
+				}
+				let read_args = ChannelManagerReadArgs::new(
+					Arc::clone(&keys_manager),
+					Arc::clone(&chain_access),
+					Arc::clone(&chain_monitor),
+					Arc::clone(&chain_access),
+					Arc::clone(&logger),
+					user_config,
+					channel_monitor_mut_references,
+				);
+				<(BlockHash, ChannelManager)>::read(&mut f, read_args)?
+			} else {
+				// We're starting a fresh node.
+				let latest_block_height = chain_access.get_height()?;
+				let latest_block_hash = chain_access.get_block_hash(latest_block_height as u64)?;
+
+				let chain_params = ChainParameters {
+					network: config.network,
+					best_block: BestBlock::new(latest_block_hash, latest_block_height),
+				};
+				let fresh_channel_manager = channelmanager::ChannelManager::new(
+					Arc::clone(&chain_access),
+					Arc::clone(&chain_monitor),
+					Arc::clone(&chain_access),
+					Arc::clone(&logger),
+					Arc::clone(&keys_manager),
+					user_config,
+					chain_params,
+				);
+				(latest_block_hash, fresh_channel_manager)
+			}
+		};
+
+		let channel_manager = Arc::new(channel_manager);
+
+		// Step 8: Give ChannelMonitors to ChainMonitor
+		for (_blockhash, channel_monitor) in channel_monitors.drain(..) {
+			let funding_outpoint = channel_monitor.get_funding_txo().0;
+			chain_monitor.watch_channel(funding_outpoint, channel_monitor).unwrap();
+		}
+
+		// TODO: Use RGS on first start, if configured
+		// Step 10: Initialize the P2PGossipSync
+		let network_graph =
+			Arc::new(util::read_network_graph(Arc::clone(&config), Arc::clone(&logger))?);
+		let gossip_sync = Arc::new(P2PGossipSync::new(
+			Arc::clone(&network_graph),
+			None::<Arc<dyn Access + Send + Sync>>,
+			Arc::clone(&logger),
+		));
+
+		//// Step 11: Initialize the PeerManager
+		let mut ephemeral_bytes = [0; 32];
+		rand::thread_rng().fill_bytes(&mut ephemeral_bytes);
+		let lightning_msg_handler = MessageHandler {
+			chan_handler: Arc::clone(&channel_manager),
+			route_handler: Arc::clone(&gossip_sync),
+		};
+
+		let peer_manager: Arc<PeerManager> = Arc::new(PeerManager::new(
+			lightning_msg_handler,
+			keys_manager.get_node_secret(Recipient::Node).unwrap(),
+			&ephemeral_bytes,
+			Arc::clone(&logger),
+			Arc::new(IgnoringMessageHandler {}),
+		));
+
+		// Step 12: Initialize routing ProbabilisticScorer
+		let scorer = Arc::new(Mutex::new(util::read_scorer(
+			Arc::clone(&config),
+			Arc::clone(&network_graph),
+			Arc::clone(&logger),
+		)));
+
+		// Step 13: Init payment info storage
+		// TODO: persist payment info to disk
+		let inbound_payments = Arc::new(Mutex::new(HashMap::new()));
+		let outbound_payments = Arc::new(Mutex::new(HashMap::new()));
```

Comment:
why separate maps ? 
is this paymentId to PaymentInfo ?

Reply:
Not sure yet how and what details we want to store exactly, but following the sample it's probably best to keep inbound and outbound separate for the time being. This is `PaymentHash` to `PaymentInfo`, as this makes it much easier to lookup payments, e.g., since incoming events don't always have the payment ID associated.